### ISSUE TITLE
Unify autograder type casing

### ISSUE THREAD

**Issue by MasterOdin** [Submitty/Submitty#1928]
From http://submitty.org/instructor/assignment_configuration:

> field: "type"
> type: string 
> value: "compilation", "file_check", or "execution"
> default value: "execution"

However, the accepted values seem to be "compilation", "FileCheck", and "execution".

Similarly, for Validator methods, one of them is "ImageDiff" which is CamelCase while the rest are capitalized via camelCase. We should try and unify all of these naming schemes to be more consistent.

**Comment by christopherturner** [comment]
Agreed @dpshelio – I'm trying to clarify this with [an update](https://github.com/Submitty/submitty.github.io/pull/48) to the [instructor assignment configuration page](https://submitty.org/instructor/assignment_configuration#specification-of-a-testcase) on the Submitty docs.

**Comment by MasterOdin** [comment]
I think the best thing here is to always convert these things to lowercase and do a string comparison against that. We can then document them however we want (preferably consistently though). We then don't break backwards-compatibility with any existing script as well as handle any capitalization typos going forward. At worst, we can just have the script when it loads the config output a warning about it, but that's probably not necessary at the moment.

**Comment by bmcutler** [comment]
related to PR #3610?
